listenbrainz/db/user_timeline_event.py

```diff
@@ -170,6 +217,53 @@ def get_recording_recommendation_events_for_feed(user_ids: List[int], min_ts: in
         return [UserTimelineEvent(**row) for row in result.fetchall()]
 
 
+def get_personal_recommendation_events_for_feed(user_id: int, min_ts: int, max_ts: int, count: int) -> List[UserTimelineEvent]:
+    """ Gets a list of personal_recording_recommendation events for specified users.
+
+    user_ids is a tuple of user row IDs.
+    """
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT user_timeline_event.id
+                 , user_timeline_event.user_id
+                 , user_timeline_event.event_type
+                 ,
+                 (
+                    SELECT jsonb_build_object(
+                        'track_name', user_timeline_event.metadata -> 'track_name',
+                        'artist_name', user_timeline_event.metadata -> 'artist_name',
+                        'release_name', user_timeline_event.metadata -> 'release_name',
+                        'recording_mbid', user_timeline_event.metadata -> 'recording_mbid',
+                        'recording_msid', user_timeline_event.metadata -> 'recording_msid',
+                        'followers_username', jsonb_agg("user".musicbrainz_id),
+                        'blurb_content', user_timeline_event.metadata -> 'blurb_content'
+                    ) AS metadata
+                    FROM jsonb_array_elements_text(user_timeline_event.metadata -> 'followers') AS arr
+                   INNER JOIN "user" ON arr.value::int = "user".id
+                 )
+                 , user_timeline_event.created
+                 , "user".musicbrainz_id as user_name
+              FROM user_timeline_event
+              JOIN "user"
```

Comment:
We will always the user name available in the view so we can just use that and avoid this join.